routers/user.js

```diff
@@ -35,30 +35,78 @@ class User {
                 lastName: context.userInfo.lastName,
                 createdAt: session_currentTime,
                 updateAt: session_currentTime
-                }
+            }
             await this.userCollection.insert(user);
+        } else {
+            user = result[0];
         }
-        else{
-            user =  result[0];
-        }
-        if (result.matchedCount < 1){
+        if (result.matchedCount < 1) {
             let error = "there is an error getting the result";
             console.error(error)
             throw new Error(error)
-          } 
+        }
         context.userInfo = {
             ...user,
             ...context.userInfo
         }
         return user
     }
-    
 
-    async updateMyUser(parm, context) {
+    async updateMyUser(params, context) {
+        let session_currentTime = getCurrentTimeYYYYMMDDSS();
+        let user
+
+        params.userInfo._id = context.userInfo._id;
+        params.userInfo.updateAt = context.userInfo.updateAt;
+        const target_obj ={
+            _id: context.userInfo._id,
+            firstName: params.userInfo.firstName,
+            lastName: params.userInfo.lastName,
+            updateAt: session_currentTime
+        }
+        const original_result = await this.userCollection.find(context.userInfo._id);
+
```

Comment:
The updated user object can be defined by merging params and the result of the find operation. For example:
`let user = await this.userCollection.find(context.userInfo._id);`
`if (!user || !Array.isArray(user) || user.length < 1) throw new Error("error message here")`
`user = { ...user[0], ...params, updateAt: getCurrentTimeYYYYMMDDSS()}`